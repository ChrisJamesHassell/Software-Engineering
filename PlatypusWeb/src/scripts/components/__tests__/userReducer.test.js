import * as actions from '../../reducers/userReducer';
import userReducer from '../../reducers/userReducer';
import { shallow, mount, render } from 'enzyme';
import React from 'react';


describe('User reducer', () =>{

    it('matches the snapshot', () =>{
        const tree = shallow(<userReducer />)
        expect(tree).toMatchSnapshot()
    })
  })

  describe('user reducer ', () => {

    it('returns the initial state', () => {
        expect(userReducer( undefined ,{} )).toEqual({});
      });
    it('UPDATE_USER', () =>{
        const startAction = {
            type: actions.UPDATE_USER 
        };
        expect(userReducer(undefined ,startAction )).toEqual({});
    });

});