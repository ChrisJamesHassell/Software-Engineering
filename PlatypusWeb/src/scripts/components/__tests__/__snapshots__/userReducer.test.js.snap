// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User reducer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <userReducer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "userReducer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "userReducer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
